client/src/pages/Bookings/Booking/EditBooking.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { IconButton, Button, Popover } from '@mui/material';
+import SettingsIcon from '@mui/icons-material/Settings';
+import { RequestStatus } from '../../../types/RequestStatus';
+
+interface Props {
+  requestId: string;
+  handleRequest: (requestId: string, status: RequestStatus) => void;
+}
+
+export default function BookingPopover({ requestId, handleRequest }: Props) {
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
+
+  return (
+    <>
+      <IconButton size="small" sx={{ marginTop: -3 }} onClick={handleClick}>
+        <SettingsIcon fontSize="small" htmlColor="lightgray" />
+      </IconButton>
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        onClose={handleClose}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'left',
+        }}
+      >
+        <Button
+          onClick={() => {
```

Comment:
An inline function in render allocates a new function instance each time render is called. Some suggest that this will cause performance issues because the garbage collector will be be invoked more often. As a result, many React developers encourage you to never use inline functions in render :)

Ditto below!